frontend/src/lib/utils/sns-aggregator-converters.utils.ts

```
@@ -334,3 +338,84 @@ const convertSnsData = ({
 
 export const convertDtoData = (data: CachedSnsDto[]): CachedSns[] =>
   data.map(convertSnsData);
+
+/**
+ * Metadata is given only if all its properties are defined.
```

Is it normal for some properties not to be defined or is this defensive programming?
Why or when would some properties not be defined?

---

I want to ask the NNS Team how much they are enforced and why they are optional.

From the type, they are optional, that's why I do this here.

What would you do if the type is optional, but they can never be so? Throw an error instead of ignoring the data?

---

I would prefer to throw an error but I would need to really understand the situation.
It's also not clear to me why we drop all the metadata if just some of the metadata is missing.
But we can go with this and see what the NNS team answers.

---

Accepting partial metadata makes sense to me. That would mean changing a few things though.

Let's better understand it and then we decide.